memory_model/src/guest_memory.rs

```diff
@@ -124,13 +125,11 @@ impl GuestMemory {
 
     /// Returns the address plus the offset if it is in range.
     pub fn checked_offset(&self, base: GuestAddress, offset: usize) -> Option<GuestAddress> {
```

Comment:
Cleaning this up is great. How about, while at it, also fixing `region_end()`. It has a misleading name, since the returned value is not actually an end (i.e. the last usable byte, `base + len - 1`), but the first non-usable byte (i.e. `base + len`).

Reply:
I favor having `region_end()` actually return the region end (i.e. last usable byte). That's because `base + end - 1` is always a valid number (when both `base` and `end` are valid), as opposed to `base + end`, which may overflow for valid values of `base` and `end`.

Reply:
I agree. I will update region_end() as you suggest.